src/esp/nav/PathFinder.cpp

```diff
@@ -253,15 +253,16 @@ bool PathFinder::build(const NavMeshSettings& bs,
   cfg.cs = bs.cellSize;
   cfg.ch = bs.cellHeight;
   cfg.walkableSlopeAngle = bs.agentMaxSlope;
-  cfg.walkableHeight = (int)ceilf(bs.agentHeight / cfg.ch);
-  cfg.walkableClimb = (int)floorf(bs.agentMaxClimb / cfg.ch);
-  cfg.walkableRadius = (int)ceilf(bs.agentRadius / cfg.cs);
-  cfg.maxEdgeLen = (int)(bs.edgeMaxLen / bs.cellSize);
+  cfg.walkableHeight = static_cast<int>(ceilf(bs.agentHeight / cfg.ch));
+  cfg.walkableClimb = static_cast<int>(floorf(bs.agentMaxClimb / cfg.ch));
+  cfg.walkableRadius = static_cast<int>(ceilf(bs.agentRadius / cfg.cs));
+  cfg.maxEdgeLen = static_cast<int>(bs.edgeMaxLen / bs.cellSize);
```

Comment:
Is "constructor cast" allowed by this rule? I.e., `int(ceilf(...))`. Because I'm using it extensively myself and magnum docs are full of it :)

Reply:
functional cast expressions are not reported by this rule but are equivalent to a c-style cast so less safe than a static_cast.

```
char *p;
long i = long(p); // will compile
long j = static_cast<long>(p); // will not compile
```

Reply:
Yes, I'm aware of the safety concerns.